src/connectors/dispatcher/impl.js

```diff
@@ -8,7 +8,17 @@ const {host, insecure, auth} = require('../../config').dispatcher;
 const Connector = require('../Connector');
 const metrics = require('../metrics');
 
-const QSOPTIONS = { encode: false };
+const FILTEROPTIONS = { encode: true };
+const FIELDOPTIONS = { encode: true, indices: false };
+
+function generateQueries (filter, fields) {
+    const queries = [
```

Comment:
So we have

1. stringifying each parameter
1. construction of array of two sets of query params
1. concating the array with results

That's an OK way to do it. A bit cleaner and simpler way to do this would be to:

1. Construct and object with both filter and fields objects nested
2. Call stringify on this object.

E.g.
```
return qs.stringify({
    filter: filter.filter,
    fields: fields.fields
}, OPTIONS)
```

Of course you could even deconstruct the values in argument definitions, etc.